src/bp/sdk/botpress.d.ts

```diff
@@ -857,8 +863,9 @@ declare module 'botpress/sdk' {
      * Call next with an error as first argument to throw an error
      * Call next with true as second argument to swallow the event (i.e. stop the processing chain)
      * Call next with no parameters or false as second argument to continue processing to next middleware
+     * Call next with the last parameter as true to mark the middleware as "skipped" in the event processing
      */
-    export type MiddlewareNextCallback = (error?: Error, swallow?: boolean) => void
+    export type MiddlewareNextCallback = (error?: Error, swallow?: boolean, skipped?: boolean) => void
```

Comment:
I know its handy here to just pass another boolean variable, but it might be the perfect time to switch from bools to a real plain datastructure.

I suggest something like:
```
type MiddlewareNextInfo = {
swallow: boolean,
skipped: boolean
}

export type MiddlewareNextCallback = (error?: Error, info?: Partial<MiddlewareNextInfo>) => void
```
in the long run, if `next()` keep evolving with others boolean parameters, 

the calls of next, instead of looking like this:
```
next(undefined, false, true, ..., false, true, true, false) // I'm exaggerating lol
```

will look like this : 
```
const info = {
skipped: true,
swallow: false,
otherFuturParam: true
}
next(undefined, info)
```

which I way more readable and also easely extensible.

However, it is a bigger reafctor to make...

What do you think about this Yann? Do you agree?

Reply:
I thought about that, but that may cause problem for custom modules since we're always going to pass an object to "swallow", so it may cause unexpected behavior.  Since there's not a lot of parameters, and we have typings, I went the easy/safe way.